enterprise/backend/src/metabase_enterprise/transforms/models/transform_run.clj

```diff
@@ -171,52 +192,42 @@
                            :nulls-last
                            :nulls-first)
         sort-column      (keyword sort_column)
-        order-by         (case sort_column
+        order-by         (case sort-column
                            :started_at [[sort-column sort-direction]]
                            :ended_at   [[sort-column sort-direction nulls-sort]]
                            [[:start_time sort-direction]
                             [:end_time   sort-direction nulls-sort]])
-        ;; Build WHERE clause conditions
-        where-conditions (cond-> []
-                           ;; transform_ids and transform_tag_ids (intersection)
-                           (and (seq transform_ids) (seq transform_tag_ids))
-                           (conj [:and
-                                  [:in :transform_id transform_ids]
-                                  [:in :transform_id {:select [:transform_id]
-                                                      :from   [:transform_transform_tag]
-                                                      :where  [:in :tag_id transform_tag_ids]}]])
-
-                           ;; Only transform_ids
-                           (and (seq transform_ids) (not (seq transform_tag_ids)))
+        where-cond       (cond-> []
+                           (some? start_time)
+                           (conj (timestamp-constraint :start_time start_time))
+
+                           (some? end_time)
+                           (conj (timestamp-constraint :end_time end_time))
+
+                           (some? run_method)
+                           (conj [:= :run_method run_method])
+
+                           (seq transform_ids)
                            (conj [:in :transform_id transform_ids])
 
-                           ;; Only transform_tag_ids
-                           (and (seq transform_tag_ids) (not (seq transform_ids)))
+                           (seq transform_tag_ids)
                            (conj [:in :transform_id {:select [:transform_id]
                                                      :from   [:transform_transform_tag]
                                                      :where  [:in :tag_id transform_tag_ids]}])
 
-                           ;; statuses condition
                            (seq statuses)
                            (conj [:in :status statuses])
 
-                           ;; is_active condition for started status
-                           (and (seq statuses)
-                                (some #(= % "started") statuses))
+                           ;; optimization: is_active condition for started status
+                           (and (= (first statuses) "started")
+                                (nil? (next statuses)))
```

Comment:
bugfix: the optimization can only be applied if no other statuses have to be accepted.